c error when passing int to fn which takes types of empty interface

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux version 5.14.6-arch1-1 (linux@archlinux) (gcc (GCC) 11.1.0, GNU ld (GNU Binutils) 2.36.1) #1 SMP PREEMPT Sat, 18 Sep 2021 16:19:35 +0000
Processor: 12 cpus, 64bit, little endian, AMD Ryzen 5 2600X Six-Core Processor
CC version: cc (GCC) 11.1.0

getwd: /home/sami
vmodules: /home/sami/.vmodules
vroot: /home/sami/v
vexe: /home/sami/v/v
vexe mtime: 2021-10-03 16:56:44
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 86a5e72.82e6d6e

Git version: git version 2.33.0
Git vroot status: weekly.2021.39.1-42-g82e6d6e5
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg /home/sami/testv.v`
```v
interface IExample {}
struct Example {
	n int
}

fn test(a IExample) {}

fn main() {
	test(Example {n: 123}) // works
	test("abc") // works
	test(true) // works
	test(123) // c error
}

```

**What did you expect to see?**

probably success, otherwise checker error


**What did you see instead?**
```
==================
/tmp/v/testv.109905451569885861.tmp.c:10526: warning: implicit declaration of function 'I_int_to_Interface_main__IExample'
/tmp/v/testv.109905451569885861.tmp.c:10526: error: cannot convert 'int' to 'struct main__IExample'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2246,7 +2246,6 @@ fn (mut g Gen) expr_with_cast(expr ast.Expr, got_type_raw ast.Type, expected_typ
 				'f64' { 'float_literal' }
 				else { got_styp }
 			}
-			println('$got_type $got_styp')
 			exp_styp := exp_sym.cname
 			mut fname := '/*$exp_sym*/I_${got_styp}_to_Interface_$exp_styp'
 			if exp_sym.info.is_generic {
@@ -6889,7 +6888,6 @@ fn (mut g Gen) interface_table() string {
 			// cctype is the Cleaned Concrete Type name, *without ptr*,
 			// i.e. cctype is always just Cat, not Cat_ptr:
 			cctype := g.cc_type(st, true)
-			println('$st, $cctype')
 			$if debug_interface_table ? {
 				eprintln(
 					'>> interface name: $ityp.name | concrete type: $st.debug() | st symname: ' +

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2242,9 +2242,9 @@ fn (mut g Gen) expr_with_cast(expr ast.Expr, got_type_raw ast.Type, expected_typ
 		} else {
 			mut got_styp := g.cc_type(got_type, true)
 			got_styp = match got_styp {
-				 'int' { 'int_literal' }
-				 'f64' { 'float_literal' }
-				 else { got_styp }
+				'int' { 'int_literal' }
+				'f64' { 'float_literal' }
+				else { got_styp }
 			}
 			println('$got_type $got_styp')
 			exp_styp := exp_sym.cname

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2240,7 +2240,13 @@ fn (mut g Gen) expr_with_cast(expr ast.Expr, got_type_raw ast.Type, expected_typ
 				got_styp)
 			g.inside_cast_in_heap--
 		} else {
-			got_styp := g.cc_type(got_type, true)
+			mut got_styp := g.cc_type(got_type, true)
+			got_styp = match got_styp {
+				 'int' { 'int_literal' }
+				 'f64' { 'float_literal' }
+				 else { got_styp }
+			}
+			println('$got_type $got_styp')
 			exp_styp := exp_sym.cname
 			mut fname := '/*$exp_sym*/I_${got_styp}_to_Interface_$exp_styp'
 			if exp_sym.info.is_generic {
@@ -6883,6 +6889,7 @@ fn (mut g Gen) interface_table() string {
 			// cctype is the Cleaned Concrete Type name, *without ptr*,
 			// i.e. cctype is always just Cat, not Cat_ptr:
 			cctype := g.cc_type(st, true)
+			println('$st, $cctype')
 			$if debug_interface_table ? {
 				eprintln(
 					'>> interface name: $ityp.name | concrete type: $st.debug() | st symname: ' +

--- a/vlib/v/tests/as_cast_literal.v
+++ b/vlib/v/tests/as_cast_literal.v
@@ -1,17 +1,12 @@
 interface IExample {}
 
-struct Example {
-	n int
-}
-
-fn test(a IExample) bool {
+fn thing(a IExample) bool {
 	return true
 }
 
-fn main() {
-	assert test(Example{123})
-	assert test(true)
-	assert test(123)
-	assert test(12.3)
-	assert test('abc')
+fn test_as_cast_with_literals() {
+	assert thing(true)
+	assert thing(123)
+	assert thing(12.3)
+	assert thing('abc')
 }

--- a/vlib/v/tests/as_cast_literal.v
+++ b/vlib/v/tests/as_cast_literal.v
@@ -1,0 +1,17 @@
+interface IExample {}
+
+struct Example {
+	n int
+}
+
+fn test(a IExample) bool {
+	return true
+}
+
+fn main() {
+	assert test(Example{123})
+	assert test(true)
+	assert test(123)
+	assert test(12.3)
+	assert test('abc')
+}

